### PR TITLE
Initial implementation of twilio-video ios sdk 2.0.0

### PR DESCRIPTION
**Pull request by cipriancaba**
Just added the new dependency and updated some of the methods.. Not everything was implemented as I didn't want to modify the API of the library as that requires a separate discussion

Please note that this is work in progress and should be seen as a kickstart for #64

### CODE REVIEW

**Comment by kcedric** [comment]
Can somebody give me write access on this project ?

**Comment by bilby91** [comment]
Hey, 

I'm out until Wednesday. Will take a look when I'm back. I would like to merge this into a v2 branch first before moving to master. 

Thanks!

**Comment by cipriancaba** [comment]
@bilby91 Makes total sense

File: ios/RCTTWSerializable.m [hunk 1]
```diff
@@ -8,18 +8,40 @@
 
 #import "RCTTWSerializable.h"
 
-@implementation TVIParticipant(RCTTWSerializable)
+@implementation TVIRemoteParticipant(RCTTWSerializable)
 
 - (id)toJSON {
-  return @{ @"identity": self.identity };
+    return @{ @"identity": self.identity };
```
Comment (bilby91) [inline ios/RCTTWSerializable.m]: Please, review indentation in changed files. The project is using 2 spaces for indentation.

File: ios/RCTTWVideoModule.m [hunk 2]
```diff
@@ -86,33 +86,37 @@ - (void)removeParticipantView:(TVIVideoView *)view identity:(NSString *)identity
...
-      for (TVIVideoTrack *videoTrack in participant.videoTracks) {
-        [videoTrack addRenderer:view];
+      for (TVIRemoteVideoTrackPublication *videoTrackPublication in participant.videoTracks) {
+        [videoTrackPublication.videoTrack addRenderer:view];
       }
     }
   }
 }
 
+- (void)logMessage:(NSString *)msg {
```
Comment (bilby91) [inline ios/RCTTWVideoModule.m]: If we are going to add logging, user's should be able to turn it on/off. Maybe we can do this in a separate PR and remove the logging for the moment and just focus on 2.0

File: ios/RCTTWVideoModule.m [hunk 3]
```diff
@@ -239,40 +246,136 @@ - (void)room:(TVIRoom *)room didFailToConnectWithError:(nonnull NSError *)error{
...
+- (void)subscribedToVideoTrack:(TVIRemoteVideoTrack *)videoTrack
+                   publication:(TVIRemoteVideoTrackPublication *)publication
+                forParticipant:(TVIRemoteParticipant *)participant {
+    
+    // We are subscribed to the remote Participant's audio Track. We will start receiving the
+    // remote Participant's video frames now.
+    
+    [self logMessage:[NSString stringWithFormat:@"Subscribed to video track for Participant %@", participant.identity]];
+    
+      [self sendEventWithName:participantAddedVideoTrack body:@{ @"participant": [participant toJSON], @"track": [videoTrack toJSON] }];
```
Comment (bilby91) [inline ios/RCTTWVideoModule.m]: Check indentation

File: ios/RCTTWVideoModule.m [hunk 4]
```diff
@@ -239,40 +246,136 @@ - (void)room:(TVIRoom *)room didFailToConnectWithError:(nonnull NSError *)error{
...
+    [self logMessage:[NSString stringWithFormat:@"Unsubscribed from audio track for Participant %@", participant.identity]];
+    [self sendEventWithName:participantRemovedAudioTrack body:@{ @"participant": [participant toJSON], @"track": [audioTrack toJSON] }];
+}
 
-- (void)participant:(TVIParticipant *)participant addedVideoTrack:(TVIVideoTrack *)videoTrack {
-  [self sendEventWithName:participantAddedVideoTrack body:@{ @"participant": [participant toJSON], @"track": [videoTrack toJSON] }];
+- (void)remoteParticipant:(TVIRemoteParticipant *)participant
+        enabledVideoTrack:(TVIRemoteVideoTrackPublication *)publication {
+    [self logMessage:[NSString stringWithFormat:@"Participant %@ enabled video track.", participant.identity]];
+//TBA      [self sendEventWithName:participantEnabledTrack body:@{ @"participant": [participant toJSON], @"track": [publication toJSON] }];
```
Comment (bilby91) [inline ios/RCTTWVideoModule.m]: I think that we can add this events without issues. We only need to add them in JS side for documentation

**Comment by cipriancaba** [comment]
I've commented the NSLog from the log method for now.. thought it might be useful for debugging atm..
Updated indentation
Will try to test tomorrow and see if the `participantEnabledTrack` and `participantDisabledTrack` work properly. Haven't tested those until now

**Comment by bilby91** [comment]
@cipriancaba Awesome! We can wait for that.

Regarding the logging, I don't have issues leaving the commented code while we are developing/testing this. When merging to master I think we need to have a proper logging or remove it and it later.

Let me know when you have tested the new events and we can merge this in v2 :)

**Comment by bilby91** [comment]
@cipriancaba ready to merge on v2?

**Comment by cipriancaba** [comment]
All clear from my side.. will still require some testing before merging into master but I strongly suggest waiting twilio sdk to come out of beta

**Comment by bilby91** [comment]
@cipriancaba Good! Let's keep this branch until Twilio releases a stable version then. After that we can merge in master 🎉

**Comment by ibash** [comment]
Hi folks!

I'm using this in a new project, and have some api suggestions. One thing that is a bit awkward is that the `TwilioVideo` component is put within the react dom, but doesn't actually display anything. I think it would be better suited as regular javascript class, that the user can instantiate and setup events to whenever they please. This would provide a cleaner api for me as I would like connect to a room outside of the react tree.

If you still wanted to provide a component it would be pretty easy to provide a thin wrapper around the regular javascript class.
